src/jsonpatcherproxy.js

```diff
@@ -143,7 +144,9 @@ var JSONPatcherProxy = (function() {
         return Reflect.deleteProperty(target, key);
       }
     });
-    return proxy;
+    /* keeping track of all the proxies to be able to revoke them later */
+    this.proxifiedObjectsMap.push(proxy);
+    return proxy.proxy;
   };
   //grab tree's leaves one by one, encapsulate them into a proxy and return
   JSONPatcherProxy.prototype._proxifyObjectTreeRecursively = function(
```

Comment:
I see my mistake here. It will emit patches to the same callback which is problematic. I think disabling the traps inside `_proxifyObjectTreeRecursively` would be better.